Updated "Use __dict__ and ignore __slots__ on classes #169

kylefawcett opened the pull request:
Fixes #168 and adds test case to prevent regression

pllim commented on CHANGES.rst:
```diff
@@ -4,6 +4,8 @@ Changes in sphinx-automodapi
 0.17.0 (unreleased)
 -------------------
 
+- Fixes fixes issue where ``__slots__`` hides class variables [#168]
```
```suggestion
- Fixes fixes issue where ``__slots__`` hides class variables [#181]
```

pllim commented:
@Nodd , do you want to have a look at the added tests?

Thanks, all!

kylefawcett commented:
@pllim Circling back around on this. I just cleaned up the slots example module to be more precisely consistent with python nomenclature and added a basic test for the example to meet the coverage thresholds. The other existing examples don't have tests so I'm not sure if you want to set a precedent. I can just pop off the last commit if you don't want tests for the slot example module.

pllim commented:
More tests is good, so no need to undo it. Thanks!

I tag some people for review because I am not familiar with slots or automodapi. But if no one responds, I can take a stab at reviewing after the holidays.

Happy holidays!

kylefawcett commented:
Happy New Year! Anyone available to review this?

pllim commented:
Hi, as people are just returning from their holiday breaks, I would give it another week or two. Thanks for your patience!

pllim commented:
There are 2 jobs that failed. Can you please rebase and see if they go away? Thanks!

pllim commented:
If rebase has nothing to do, please let me know, then I can  restart those jobs directly. Thanks!

bsipocz reviewed (approved):
I'm not familiar with __slots__ either, but dug up the history of this part of the code. It was added many years ago in https://github.com/astropy/sphinx-automodapi/pull/53, along with test cases, which still pass after this PR.

So I would say go ahead and merge this.

bsipocz commented:
Thank you @kylefawcett!
